### DISCUSSION TITLE
Ente Auth iOS Autofill

### DISCUSSION THREAD

**Discussion by gitgizmo7** [ente-io/ente#3738] [Enhancements]
With the release of iOS 18, TOTP codes can now be autofilled, and some third-party password managers have been updated to support this:
![image](https://github.com/user-attachments/assets/82d6465a-7de1-486d-aab3-34d1d8933d90)
Could Ente Auth be updated to support this? You can select up to three autofill sources, so even users who already use autofill with their password manager would be able to use this if it were added.

**Comment by mallomar** [comment]
Just was looking for this feature as well.

**Comment by harrison-labs** [comment]
I agree that this would be incredible useful.

**Comment by Guztaver** [comment]
Looking for this feature too

**Comment by nicholas-alonzo** [comment]
This would be excellent!

**Comment by Vaibhav-Lodha** [comment]
This would be a great addition to keep things seamless

**Comment by just-an-atom** [comment]
This would be amazing! I saw in the setting you can choose Ente Auth but it doesn't seem to work?

**Comment by Kradirhamik** [comment]
Would like this as well please!

**Comment by kianrafiee** [comment]
Could we get macOS support as well?

**Comment by wwcanoer** [comment]
Yes please! I just assumed that it would autofill but I could not find a way. (to copy the code, double click).  My first attempt with TOTP and I am not impressed.  Right now, I only have one site, but once there's 50, then I'll have to find and open the Ente Auth, scroll to find the target app, copy and paste.  With SMS 2FA the code will pop up on the screen so, just wait, memorize and type.

I have no idea if other TOTPs autofill or not.  I chose this one since can sync iphone/android/windows and had a good reputation, open source, no tracking.

**Comment by wwcanoer** [comment]
On Discord: "yes, iOS 18 does allow you to autofill. We haven’t spend much time in exploring how much of effort this will be considering that the app is built with flutter, and there seems to be some nuances around using the iOS credential provider service in flutter. 

As of now, everyone in the team is fully focused on Photos app, and we don’t have bandwidth to explore this in the next few of months."
-- Neeraj — 1/30/25, 11:38 AM 
https://discord.com/channels/948937918347608085/1051153671985045514/1334563439275675648

**Comment by aedelred** [comment]
I too am in need of code autofill on iOS.

**Comment by D-i9Te7-nQ** [comment]
Would appreciate this feature added.

It’s great codes can be setup using Ente in Autofill setting on iOS but it would be really nice to add Ente as Autofill From option for 2FA codes; simpler than going into app, copying code, then going back to login page and pasting 2FA code.

Thanks

**Comment by Damariobros** [comment]
I feel like, at the very least, it should support manual autofill. Like, say, select the autofill option on the keyboard, select Ente, search your accounts, and then double-tap the account otp you want to autofill.

This would save people from situations where they enter username and password, go to Ente, copy code, go back, page or app has refreshed, enter username and password again, code has expired, go to Ente, the website or app refreshes again, enter username and password again, code is expired again…

**Comment by KaungZinLin** [comment]
Would love to get this feature!

**Comment by viswagfx** [comment]
this feature would be very helpfull

**Comment by ErikZiman** [comment]
Would love to see this feature

**Comment by MehdiMamas** [comment]
WOULD LOOOOOVE THIS FEATURE

**Reply by MehdiMamas** [reply]
in mac as well :D

**Comment by iamabz** [comment]
Any updates?

**Comment by hamburger2506** [comment]
OMG please I pray 🙏

**Comment by Damariobros** [comment]
Once it's fleshed out, maybe a way to exclude certain groups of TOTP accounts from autofill just to hide them?

**Comment by anotherminor** [comment]
Would love to see this feature please!

**Comment by Kradirhamik** [comment]
Please add this!

**Comment by corentin-hannoye** [comment]
+1

**Comment by p5lukas** [comment]
This is a must! :)
